### PR TITLE
fix(worker): Disallow importing non-Workflow @temporalio packages

### PR DESCRIPTION
**Pull request by lorensr**
Following up on https://github.com/temporalio/sdk-typescript/pull/645#issuecomment-1158326034

### CODE REVIEW

File: packages/worker/src/workflow/bundler.ts [hunk 1]
```diff
@@ -11,6 +11,12 @@ import { DefaultLogger, Logger } from '../logger';
 
 export const allowedBuiltinModules = ['assert'];
 export const disallowedBuiltinModules = builtinModules.filter((module) => !allowedBuiltinModules.includes(module));
+export const disallowedModules = [
+  ...disallowedBuiltinModules,
+  '@temporalio/activity',
+  '@temporalio/client',
+  '@temporalio/worker',
```
Comment (bergundy) [inline packages/worker/src/workflow/bundler.ts:18]: Maybe add non-workflow-common, the opentelemetry client and worker libs and most of the testing stuff?

**Review by bergundy — APPROVED** [review]
Approved with a comment, thanks @lorensr

File: packages/worker/src/workflow/bundler.ts (hunk 1 above)
Comment (lorensr) [reply packages/worker/src/workflow/bundler.ts:18]: 👍 I like when I remember to use `jq`

```
$ cat packages/*/package.json | jq .name
"@temporalio/activity"
"@temporalio/client"
"@temporalio/common"
"@temporalio/core-bridge"
"@temporalio/create"
"@temporalio/docs"
"@temporalio/interceptors-opentelemetry"
"@temporalio/internal-non-workflow-common"
"@temporalio/internal-workflow-common"
"temporalio"
"@temporalio/proto"
"@temporalio/test"
"@temporalio/testing"
"@temporalio/worker"
"@temporalio/workflow"
```

**Review by vkarpov15 — APPROVED** [review]
Nice :+1:

**Review by bergundy — APPROVED** [review]

**Comment by bergundy** [comment]
Looks like CI found a real issue, nice to see that the fix works.

**Comment by bergundy** [comment]
@lorensr we should probably see what's importing from non-workflow-common and fix it.

**Comment by lorensr** [comment]
@bergundy this file imported from `otel.ts` which was located in `internal-non-workflow-common`. I moved it to `common` (not `internal-workflow-common`, since it uses `defaultDataConverter` which is in common).

**Comment by bergundy** [comment]
I'd make it so otel api isn't required for workflow context

**Comment by bergundy** [comment]
This would require some code duplication but I'd rather have that then the extra dependency

**Comment by lorensr** [comment]
For bundle size sake? Will it be included if the user doesn't import from `@temporalio/interceptors-opentelemetry/lib/workflow`?

**Comment by bergundy** [comment]
Hmm.. nevermind, it's okay, we don't export `lib/otel` from `common/index`

**Pull request merged**